src/client.cpp

```diff
@@ -908,6 +905,56 @@ void CClient::Stop()
     SignalLevelMeter.Reset();
 }
 
+/// @method
+/// @brief Connects to strServerAddress
+/// @emit Connecting (strServerName) if the client wasn't running and SetServerAddr was valid.
+///       Use to set CClientDlg to show being connected
+/// @emit ConnectingFailed (error) if an error occurred
+///       Use to display error message in CClientDlg
+/// @param strServerAddress - the server address to connect to
+/// @param strServerName - the String argument to be passed to Connecting()
+void CClient::Connect ( QString strServerAddress, QString strServerName )
+{
+    try {
+        if ( !IsRunning() )
+        {
+            // Set server address and connect if valid address was supplied
+            if ( SetServerAddr ( strServerAddress ) )
+            {
+                Start();
+                emit Connecting ( strServerName );
+            }
+            else {
+                throw CGenErr ( "Received invalid server address. Please check for typos in the provided server address." );
+            }
+        }
+    }
+    catch ( const CGenErr& generr )
+    {
+        Disconnect();
+        emit ConnectingFailed ( generr.GetErrorText() );
+    }
+}
+
+/// @method
+/// @brief Disconnects client. If the client is not running, it just stops Sound
+/// @emit Disconnected
+///       Use to set CClientDlg to show not being connected
+void CClient::Disconnect()
+{
+    if ( IsRunning() )
+    {
+        Stop();
+        emit Disconnected();
+    }
+    else
+    {
+        // make sure sound is stopped in any case
+        Sound.Stop();
```

Comment:
Maybe "Sound" should be listening to Client.Disconnected and its OnDisconnected handler should Stop?

Reply:
Sound.Stop() is driver specific, I think.
For ASIO, for example it calls ASIOStop(), iOS calls AudioOutputUnitStop(), JACK nothing, Android closeStreams(). But the important part is that all call CSoundBase::Stop().

In any way, CClient::Stop adds the 100ms timer additionally and sends a disconnect message to the server. Not sure if that's a problem (I'd rather think it isn't)

Reply:
I've for now still included a small method.

Reply:
... And removed it...